packages/zoe/src/contracts/helpers/zoeHelpers.js

```diff
@@ -0,0 +1,119 @@
+import harden from '@agoric/harden';
+import { assert, details } from '@agoric/assert';
+import { sameStructure } from '@agoric/same-structure';
+
+export const defaultRejectMsg = `The offer was invalid. Please check your refund.`;
+export const defaultAcceptanceMsg = `The offer has been accepted. Once the contract has been completed, please check your payout`;
+
+const getKeys = obj => harden(Object.getOwnPropertyNames(obj || {}));
+
+export const makeZoeHelpers = zoe => {
+  const { roles } = zoe.getInstanceRecord();
+  const amountMaths = zoe.getAmountMathsForRoles(roles);
+  const zoeService = zoe.getZoeService();
+
+  const rejectOffer = (inviteHandle, msg = defaultRejectMsg) => {
+    zoe.complete(harden([inviteHandle]));
+    throw new Error(msg);
+  };
+
+  // Compare the keys of actual with expected keys and reject offer if
+  // not sameStructure.
+  const rejectIf = (
+    inviteHandle,
+    actual,
+    expectedKeys,
+    msg = defaultRejectMsg,
+    // eslint-disable-next-line consistent-return
+  ) => {
+    if (expectedKeys !== undefined) {
+      if (!sameStructure(getKeys(actual), expectedKeys)) {
+        return rejectOffer(inviteHandle, msg);
+      }
+    }
+  };
+  // compare actual checks to expected keys. If expectedKeys is
+  // undefined, return true trivially.
+  const check = (inviteHandle, actual, expectedKeys) => {
+    if (expectedKeys === undefined) {
+      return true;
+    }
+    return sameStructure(getKeys(actual), expectedKeys);
+  };
+  const helpers = harden({
```

Comment:
Yup, I'll add tests. Dominique will be helping document them on the doc site. I think we decided not to do additional jsdoc comments though, because Michael is working on turning the chainmail docs into TypeScript interfaces